geps/gep-3793/index.md

```diff
@@ -0,0 +1,144 @@
+# GEP-3793: Default Gateways
```

Comment:
This is likely an awful idea, but I'll throw it out there as something that may at least merit some thoughts in "alternatives considered" in the next phase of the GEP. Instead of "Default Gateways", we could allow individual Gateway (Listeners) to be configured with "greedy" behavior. That might mean that they would claim all Routes that didn't have a parentRef specified, or it might mean that Routes could specify (and maybe did by default) that they should be picked up by any greedy Listeners. Just one example of a solution here not strictly requiring individual Gateways to be labeled as "default".

Reply:
This suggestion is appealing as it makes the initial configuration easy for the cluster operator and for the application developer, and it works well for small clusters.  However, something I've seen happen time and again is that a cluster starts with a single default ingress-controller with a single proxy deployment and load balancer which gradually takes on additional workload as the cluster grows until it reaches a breaking point at which the proxy deployment has become overburdened.  At that point, it can be difficult to shift workload after the fact to different proxies or load-balancers.  I am wary of designing the feature in such a way as to guide people towards brick walls.

Reply:
@kflynn I recommend adding this as a note in the GEP, perhaps in "alternatives considered" as suggested, and then we can consider this resolved for this first pass.